cmake/public/cuda.cmake

```diff
@@ -510,9 +323,26 @@ elseif (CUDA_VERSION VERSION_EQUAL 8.0)
 endif()
 
 # setting nvcc arch flags
-caffe2_select_nvcc_arch_flags(NVCC_FLAGS_EXTRA)
+if ((NOT EXISTS ${TORCH_CUDA_ARCH_LIST}) AND (DEFINED ENV{TORCH_CUDA_ARCH_LIST}))
+  message(WARNING
+      "In the future we will require one to explicitly pass "
```

Comment:
from the user's commands, i.e. `python setup.py install`, they do not specify cmake variables (or dont have a straight-forward way to do so).

We can probably make setup.py to read the environment variable TORCH_CUDA_ARCH_LIST and then pass it in to the cmake manually.

Reply:
Yep. setup.py shall take on the task of passing all proper env variables to cmake - and from the user front they can continue using the env behavior if needed.